### PR TITLE
refactor: implement prom remote query by convert to datafusion plan directly

### PR DESCRIPTION
**Pull request by jiacai2050**
## Which issue does this PR close?

Related to #855

## Rationale for this change
 
Currently prom requests is implemented based on sql, but there are some shortcomings:

1. Timestamp is hard-coded
2. Can't select column other than `value`

## What changes are included in this PR?

- Convert prom request to datafusion plan directly 
- Add special label `__ceresdb_field__` to set field column other than `value`

## Are there any user-facing changes?

No

## How does this change test

CI and manually

### CODE REVIEW

File: query_frontend/src/promql/remote.rs [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+/// ```
+pub fn remote_query_to_plan<P: MetaProvider>(
+    query: Query,
+    meta_provider: ContextProviderAdapter<'_, P>,
+) -> Result<RemoteQueryPlan> {
+    let (metric, field, mut filters) = normalize_matchers(query.matchers)?;
+    let table_ref = meta_provider
+        .table(TableReference::parse_str(&metric))
+        .context(MetaProviderError {
+            msg: "failed to find table".to_string(),
```
Comment (chunshao90) [inline query_frontend/src/promql/remote.rs:51]:
```suggestion
            msg: "Failed to find table".to_string(),
```

File: query_frontend/src/promql/remote.rs [hunk 2]
```diff
@@ -0,0 +1,249 @@
+// Copyright 2023 CeresDB Project Authors. Licensed under Apache-2.0.
+
+//! This module convert Prometheus remote query to datafusion plan
```
Comment (chunshao90) [inline query_frontend/src/promql/remote.rs:3]:
```suggestion
//! This module convert Prometheus remote query to datafusion plan.
```

File: query_frontend/src/promql/error.rs [hunk 3]
```diff
@@ -0,0 +1,53 @@
+// Copyright 2023 CeresDB Project Authors. Licensed under Apache-2.0.
+
+use ceresdbproto::prometheus::sub_expr::OperatorType;
+use datafusion::error::DataFusionError;
+use snafu::{Backtrace, Snafu};
+
+#[allow(clippy::large_enum_variant)]
+#[derive(Debug, Snafu)]
+#[snafu(visibility = "pub")]
```
Comment (chunshao90) [inline query_frontend/src/promql/error.rs:9]: Necessary?

File: query_frontend/src/promql/remote.rs [hunk 4]
```diff
@@ -0,0 +1,249 @@
...
+        .filter(conjunction(filters).expect("at least one filter(timestamp)"))?
+        .project(projection)?
+        .sort(default_sort_exprs(timestamp_col_name))?;
+
+    let df_plan = builder.build().context(BuildPlanError)?;
+    let tables = Arc::new(
+        meta_provider
+            .try_into_container()
+            .context(MetaProviderError {
+                msg: "failed to find meta",
```
Comment (chunshao90) [inline query_frontend/src/promql/remote.rs:76]:
```suggestion
                msg: "Failed to find meta",
```

File: query_frontend/src/promql/error.rs (hunk 3 above)
Comment (jiacai2050) [reply query_frontend/src/promql/error.rs:9]: If not pub, then other module(like remote.rs) can't use it.

**Review by chunshao90 — APPROVED** [review]
LGTM

**Pull request merged**